### PR TITLE
Support migrate to Python 3.12

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

Support migrate to Python 3.12. Need more examples testing.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/app/utils.py [hunk 1]
```diff
@@ -98,10 +98,14 @@ def create_admin_server(fl_server: FederatedServer, server_conf=None, args=None,
 
 
 def version_check():
-    if sys.version_info >= (3, 11):
-        raise RuntimeError("Python versions 3.11 and above are not yet supported. Please use Python 3.8, 3.9 or 3.10.")
+    if sys.version_info >= (3, 13):
```
Comment (yanchengnv) [inline nvflare/private/fed/app/utils.py:101]: Is this all we need to do? We need to run all examples and CICD tests.

**Review by yanchengnv — COMMENTED** [review]
Have we run enough tests? Are there other dependencies (e.g. pytorch, grpc) that may need changes?

File: nvflare/private/fed/app/utils.py (hunk 1 above)
Comment (yhwen) [reply nvflare/private/fed/app/utils.py:101]: I ran a couple of examples, our implementations are compatible with the Python 3.12. Yes, we need to run all the examples to certify.

Comment (yhwen) [reply nvflare/private/fed/app/utils.py:101]: @chesterxgchen Last time is migration from 3.8 to 3.10. We did make the changes to solve the in-compatible issues. Those changes already included in the latest code bases.

Comment (rfilgueiras) [reply nvflare/private/fed/app/utils.py:101]: When installing the current 2.5.0rc8 on my environment I got [this protobuf version](https://pypi.org/project/protobuf/5.28.0/).

It supports Python 3.12 :tada:

**Review by YuanTingHsieh — APPROVED** [review]

**Comment by YuanTingHsieh** [comment]
Thanks, discussed with Yuhong and we have ported the change to #2951

**Pull request closed without merging**